hilti/toolchain/src/compiler/driver.cc

```diff
@@ -682,14 +666,29 @@ Result<Nothing> Driver::run() {
 }
 
 Result<Nothing> Driver::transformUnits() {
-    if ( ! _driver_options.global_optimizations )
-        return Nothing();
+    if ( _driver_options.global_optimizations ) {
+        HILTI_DEBUG(logging::debug::Driver, "performing global transformations");
 
-    HILTI_DEBUG(logging::debug::Driver, "performing global transformations");
+        GlobalOptimizer opt(&_hlts, _ctx);
 
-    GlobalOptimizer opt(&_hlts, _ctx);
+        opt.run();
+    }
+
+    if ( _driver_options.output_hilti ) {
```

Comment:
Not sure I like having the printing inside `transformUnits()`. How about moving it up one level to between the calls to `transformUnits()` and `codegenUnits()`?